test/Exception.js

```diff
@@ -174,14 +161,46 @@ describe('Exception', () => {
 		})
 	})
 
+	it('should recover from error boundary in the event of an exception', () => {
+		const target = document.createElement('div')
+		const Primary = props => {
+			const [error, dispatch] = useState('')
+
+			return h(Boundary, {fallback: error => dispatch(error.message)}, error ? error : h(Secondary))
+		}
+		const Secondary = props => {
+			throw 1
+		}
+
+		render(h('div', {}, h(Primary)), target, (current) => {
+			assert.html(current, '<div>1</div>')
+		})
+	})
+
+	it('should throw within promise', (done) => {
+		const target = document.createElement('div')
+		const stack = assert.spyr(console, 'error')
+		const Primary = props => {
+			return stack.current = Promise.reject(1)
+		}
+
+		render(h(Primary), target, () => {}).then((current) => {
```

Comment:
I'd like to see another test that tests the second parameter to `then()`, to make sure https://github.com/dyo/dyo/issues/106 is covered

Reply:
That is handled by L#196 and L#213.